### ISSUE TITLE
PID/Power model to control brake resistance

### ISSUE THREAD

**Issue by jovial** [fluxoid-org/CyclismoProject#8]
Been mucking around with sycning the model to the wheel speed. Goes a
little something like this:
- set starting resistance (10% + gradient) (some what arbitrary)
- wait for wheel speed to reach steady state (test : average
  acceleration +/- 0.28 m/s^2 (3+ samples))
- sync model to this speed
- let PID take control 

What do you reckon? Is this madness? One thing it does do is to get rid
of the initial resistance fluctuations. The PID does fairly well at keep
ing speed close to that of model (once synced), as long as you are above
a certain speed. I don't think the trainer does too well at simulating
speeds below 18 km/h.

The disadvantages are that the acceleration to the steady state speed is
some what contrived and you don't know what this will be. At present the
initial resistance is fairly high to try an get a lower sync speed. This
is noticeable in the feel when the resistance drops off once synced.

An alternative is to enforce a minimum setpoint (predicted) value. Maybe
with high gain coefficients to get rapid convergence to that of the
model. Would need to select an appropriate minimum speed though (kind of
why I thought syncing would be better as that would depend on the power
you are outputting).

Setting a resistance value manually, the brake can react within a
second. However, to get a large enough initial value with the PID
requires a large proportional term (to kick in quickly). This didn't
work too well without the minimum speed limit as it aggressively tried
to reduce the speed to near zero. A minimum speed however, would reduce
this initial kick.

Basically it boils down to picking an appropriate minimum speed or an
appropriate initial resistance. 

Any thoughts?

Will
